xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -1159,8 +1159,7 @@ bool CDVDInputStreamNavigator::SetAngle(int angle)
 
 bool CDVDInputStreamNavigator::GetCurrentButtonInfo(CDVDOverlaySpu* pOverlayPicture, CDVDDemuxSPU* pSPU, int iButtonType)
 {
-  int alpha[2][4];
-  int color[2][4];
+  int colorAndAlpha[4][4];
```

Comment:
maybe use a c++ `std::array<int, 4 * 4> colorAndAlpha;` and then access its elements via `colorAndAlpha.at(x * 4 + y) = ...` as this provides bounds checking. well not sure if it's worth for such a small thing, but...

Reply:
Compared to the ridiculous amount of time I've spent on this I'll gladly try to implement it :)